File: amarath/src/middlewares/LoggerMiddleware.ts
import { Logger } from "@logtape/logtape";
import Elysia from "elysia";

export function NewLoggerMiddleware(logger: Logger) {
  return new Elysia()
    .resolve({ as: "global" }, () => {
      const start = performance.now();
      return { start };
    })
    .onAfterHandle({ as: "global" }, ({ request, path, start, set }) => {
      const duration = performance.now() - start;
      logger.info(
        `${request.method} ${path}: ${set.status} - ${duration.toFixed(2)}ms`,
      );
    });
}

File: amary/src/middlewares/logger_middleware.go
package middlewares

import (
	"time"

	"github.com/gin-gonic/gin"
)

func NewLoggerMiddleware(lg Logger) gin.HandlerFunc {
	return func(ctx *gin.Context) {
		start := time.Now()

		ctx.Next()

		elapsed := time.Since(start)
		lg.Infoln(ctx.Request.Method, ctx.Request.URL.Path, ctx.Writer.Status(), elapsed.Abs().String())
	}
}
